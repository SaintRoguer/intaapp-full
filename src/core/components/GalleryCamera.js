import {Container, Content, Spinner} from 'native-base';
import React, {Component} from 'react';
import {StyleSheet} from 'react-native';
import {mainThemeColor, galleryCameraButtons} from '../../configuration';
import {
  HomeCard,
  withImagePicker,
  withImageProcessing,
  withImageHandler,
} from '../../shared';
import {ImageModel} from '../../shared/models/ImageModel.js';

class GalleryCamera extends Component {
  constructor(props) {
    super(props);
    this.state = {
      loading: false,
    };
  }

  render() {
    return (
      <Container>
        <Content contentContainerStyle={styles.container}>
          {this.state.loading ? (
            <Spinner color={mainThemeColor(1)} />
          ) : (
            <>{this.getGalleryCameraButtons()}</>
          )}
        </Content>
      </Container>
    );
  }

  getGalleryCameraButtons() {
    return galleryCameraButtons.map((button, index) => (
      <HomeCard
        key={index}
        onPress={this.launch(button.type)}
        icon={button.icon}
        text={button.text}
      />
    ));
  }

  launch = picker => async () => {
    try {
      this.setState({loading: true});
      const imageResponse = await this.props.imageHandler.pickImage()(picker);
      this.props.navigation.navigate('Imagen', imageResponse);
    } finally {
      this.setState({loading: false});
    }
  };
}

export default withImageHandler(GalleryCamera);
const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
});
