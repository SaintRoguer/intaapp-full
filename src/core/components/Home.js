import {Container, Content, Spinner} from 'native-base';

import React, {Component} from 'react';
import {StyleSheet, Alert} from 'react-native';
import {mainThemeColor, homeButtons} from '../../configuration';
import {HomeCard, ImagePickerService, ImageProcessor} from '../../shared';

export class Home extends Component {
  picker;
  imageProcessor;

  constructor(props) {
    super(props);
    this.picker = ImagePickerService.getInstance();
    this.imageProcessor = ImageProcessor.getInstance();
    this.state = {
      loading: false,
    };
  }

  render() {
    return (
      <Container>
        <Content contentContainerStyle={styles.container}>
          {this.state.loading ? (
            <Spinner color={mainThemeColor(1)} />
          ) : (
            <>{this.getHomeButtons()}</>
          )}
        </Content>
      </Container>
    );
  }

  getHomeButtons() {
    return homeButtons.map((button, index) => (
      <HomeCard
        key={index}
        onPress={this.launch(button.type)}
        icon={button.icon}
        text={button.text}
      />
    ));
  }

  launch = picker => async () => {
    this.props.navigation.navigate('GalleryCamera', {});
  };
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    // backgroundColor: '#20d2bb',
  },
});
