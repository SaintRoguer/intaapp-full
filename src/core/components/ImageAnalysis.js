//@ts-check
import React, {useState} from 'react';
import {StyleSheet, View} from 'react-native';
import {SaveImage, ImageView, Tabs, ImageNotes} from '../../shared';
import {withImageProcessing} from '../../shared';

function ImageAnalysis({route: {params}, imageProcessor}) {
  const onSave = params.onSave;
  const [originalImage, setOriginalImage] = useState(params.originalImage);
  const [processedImage, setProcessedImage] = useState(params.processedImage);
  const [percentages, setPercentages] = useState(params.percentages);
  const [note, setNote] = useState('');

  const updateOriginalImage = async newUri => {
    let {
      percentageGreen,
      percentageYellow,
      percentageNaked,
      img,
    } = await imageProcessor.processImage('file://' + newUri);
    setPercentages({
      percentageGreen,
      percentageYellow,
      percentageNaked,
    });

    setOriginalImage(prev => {
      const newOriginal = prev;
      newOriginal.setUri('file://' + newUri);
      return newOriginal.clone();
    });
    setProcessedImage(prev => {
      const newProcessed = prev;
      newProcessed.setData(img);
      return newProcessed.clone();
    });
  };

  const saveImage = async () => {
    return await onSave(percentages, originalImage.getSource(), note);
  };

  const getImageView = () => (
    <ImageView
      shouldRotate={params.shouldRotate}
      updateOriginalImage={updateOriginalImage}
      originalImage={originalImage}
      staticOriginal={params.originalImage.clone()}
      processedImage={processedImage}
      percentages={percentages}
    />
  );

  return (
    <View style={styles.container}>
      {onSave ? (
        <Tabs
          FirstScreen={getImageView}
          SecondScreen={() => <ImageNotes value={note} updateNote={setNote} />}
          secondTitle="Notas"
          firstTitle="AnÃ¡lisis"
        />
      ) : (
        getImageView()
      )}
      {onSave ? <SaveImage onSave={saveImage} /> : null}
    </View>
  );
}
export default withImageProcessing(ImageAnalysis);

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  information: {
    flex: 1,
  },
  switchContainer: {
    width: '100%',
    flex: 1,
    justifyContent: 'flex-end',
    alignItems: 'center',
  },
  switch: {
    width: '80%',
    zIndex: 99,
  },
});
