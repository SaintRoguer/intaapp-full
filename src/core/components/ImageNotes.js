import React from 'react';
import {View} from 'react-native';

export function ImageNotes() {
  return <View />;
}
