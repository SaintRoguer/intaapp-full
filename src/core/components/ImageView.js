import React, {Component} from 'react';
import {StyleSheet, View} from 'react-native';
import SwitchSelector from 'react-native-switch-selector';
import {options, switchConfig} from '../../configuration';
import {
  ImageProcessor,
  ImageWithAdjustment,
  FirebaseService,
  CloudinaryService,
} from '../../shared';
import {Percentages} from './Percentages';

export class ImageView extends Component {
  imageProcessor;
  originalImage;
  cloudinaryService;

  constructor(props) {
    super(props);
    this.state = {
      showOriginal: false,
      ...this.props.route.params,
    };
    this.originalImage = this.state.originalImage.clone();
    this.imageProcessor = ImageProcessor.getInstance();
    this.cloudinaryService = CloudinaryService.getInstance();
    this.firebaseService = FirebaseService.getInstance();
  }

  render() {
    return (
      <View style={styles.container}>
        <View style={styles.switchContainer}>
          <SwitchSelector
            style={styles.switch}
            options={options}
            onPress={value =>
              this.setState({
                showOriginal: value,
              })
            }
            {...switchConfig}
          />
        </View>
        <ImageWithAdjustment
          onImageAdjusted={this.updateOriginalImage}
          imageToEdit={this.originalImage}
          shouldRotate={this.props.route.params.shouldRotate}
          imageToShow={this.getImage()}
        />
        <Percentages
          percentageGreen={this.state.percentageGreen}
          percentageYellow={this.state.percentageYellow}
          percentageNaked={this.state.percentageNaked}
        />
      </View>
    );
  }

  updateOriginalImage = async newUri => {
    let nativeReponse = await this.imageProcessor.processImage(
      'file://' + newUri,
    );

    this.setState(async prevState => {
      let newState = {
        ...prevState,
      };
      newState.originalImage.setUri('file://' + newUri);
      newState.processedImage.setData(nativeReponse.img);
      newState.percentageGreen = nativeReponse.percentageGreen;
      newState.percentageYellow = nativeReponse.percentageYellow;
      newState.percentageNaked = nativeReponse.percentageNaked;

      // // const uri = await this.cloudinaryService.uploadPhoto(
      // //   `data:image/png;base64,${newState.processedImage.getData()}`,
      // // );
      // const uri = await this.cloudinaryService.uploadPhoto(
      //   newState.originalImage.getUri(),
      // );
      // if (uri != null) {
      //   this.firebaseService.uploadPhoto(uri);
      // }
      return newState;
    });
  };

  getImage() {
    let sourceOfImage = this.state.showOriginal
      ? this.state.originalImage
      : this.state.processedImage;
    return sourceOfImage;
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
  },
  information: {
    flex: 1,
  },
  switchContainer: {
    width: '100%',
    flex: 1,
    justifyContent: 'flex-end',
    alignItems: 'center',
  },
  switch: {
    width: '80%',
    zIndex: 99,
  },
});
