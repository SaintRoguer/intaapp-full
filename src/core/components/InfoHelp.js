//@ts-check
import React, {useState, useEffect} from 'react';
import {StyleSheet, View, Text} from 'react-native';
import {HeaderBackButton} from '@react-navigation/stack';

function InfoHelp({navigation}) {
  useEffect(() => {
    navigation.setOptions({
      headerLeft: () => (
        <HeaderBackButton
          tintColor="white"
          //   onPress={() => {
          //     navigation.navigate('Main', {
          //       showInfo: false,
          //     });
          //   }}
        />
      ),
    });
  }, []);

  return (
    <View style={styles.container}>
      <Text>INfo</Text>
    </View>
  );
}
export default InfoHelp;

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});
