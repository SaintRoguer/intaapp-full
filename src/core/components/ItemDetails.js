//@ts-check
import React, {useState, useEffect} from 'react';
import {Text, StyleSheet, View} from 'react-native';
import {
  withAlertService,
  withFirebase,
  Tabs,
  uniqueId,
  GridWithNewButton,
  ImagesTaken,
  BottomRightButton,
  Info,
  withImageHandler,
} from '../../shared/';

function ItemDetails({
  firebaseService: firebase,
  alertService: alerts,
  navigation,
  imageHandler,
}) {
  const [images, setImages] = useState([]);
  const [pasturas, setPasturas] = useState([]);
  const [itemDetail, setItemDetail] = useState();
  const {item} = navigation.state.params;
  useEffect(() => {
    firebase
      .getDataFromInnerId('lotesDetails')(item.id)
      .then(item => {
        setItemDetail(item);
        setImages(item.images);
        setPasturas(item.pasturas);
      });
  }, [item.id]);
  const handleImage = imageHandler.pickImage({
    collectionName: 'lotesDetails',
    itemId: item.id,
  });
  const noop = () => {};
  return (
    <View style={styles.detailsContainer}>
      <Info item={itemDetail} />
      <Tabs
        firstTitle="Pasturas"
        secondTitle="Imagenes"
        FirstScreen={() => (
          <GridWithNewButton
            title=""
            data={pasturas}
            onEntryClick={noop}
            onNewClick={noop}
            onDeleteEntry={noop}
          />
        )}
        SecondScreen={() => <ImagesTaken images={images} loteId={item.id} />}
      />
      <BottomRightButton
        withBackground={true}
        buttons={[
          {
            name: 'upload',
            type: 'FontAwesome5',
            onPress: async () => {
              const imageResponse = await handleImage('Gallery');
              navigation.navigate('Imagen', imageResponse);
            },
          },
          {
            name: 'camera-retro',
            type: 'FontAwesome5',
            onPress: async () => {
              const imageResponse = await handleImage('Camera');
              navigation.navigate('Imagen', imageResponse);
            },
          },
        ]}
      />
    </View>
  );
}

const styles = StyleSheet.create({
  detailsContainer: {
    height: '100%',
  },
  description: {
    fontSize: 21,
    textAlign: 'center',
  },
});

export default withImageHandler(withAlertService(withFirebase(ItemDetails)));
