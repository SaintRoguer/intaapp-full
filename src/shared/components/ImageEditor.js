import {Button} from 'native-base';
import React, {Component} from 'react';
import {mainThemeColor} from '../../configuration';
import {Dimensions, StyleSheet, Text, View} from 'react-native';
import Popover from 'react-native-popover-view';
import {imageAdjusts} from '../../configuration';
import CustomImage from './CustomNativeImage';
import {Sliders} from './Sliders';
import {BottomRightButton} from './BottomRightButton';

export class ImageEditor extends Component {
  constructor(props) {
    super(props);
    this.state = {
      image: this.props.image,
      brightness: imageAdjusts[0].defaultValue,
      saturation: imageAdjusts[1].defaultValue,
      contrast: imageAdjusts[2].defaultValue,
      saveImg: false,
    };
  }

  render() {
    return (
      <Popover
        onCloseComplete={this.resetState}
        popoverStyle={styles.popover}
        isVisible={this.props.showOver}>
        <View style={styles.imageContainer}>
          <BottomRightButton
            type="Ionicons"
            name="refresh"
            onPress={this.resetAdjustment}
          />
          <CustomImage
            onSave={this.save}
            saveImage={this.state.saveImg}
            resizeMode="contain"
            style={styles.image}
            src={{
              source: [
                {
                  uri: this.props.image.getSource(),
                },
              ],
              width: this.props.image.getWidth(),
              height: this.props.image.getHeight(),
            }}
            brightness={this.state.brightness}
            saturation={this.state.saturation}
            contrast={this.state.contrast}
          />
        </View>
        <View style={styles.settingsContainer}>
          <Sliders
            updateValue={this.updateValue}
            contrast={this.state.contrast}
            brightness={this.state.brightness}
            saturation={this.state.saturation}
          />
        </View>
        <View style={styles.buttonsContainer}>
          <Button style={styles.button} onPress={this.props.onClose} light>
            <Text style={styles.buttonText}>Cancelar</Text>
          </Button>
          <Button style={styles.button} onPress={this.saveImg} primary>
            <Text style={styles.buttonText}>Confirmar</Text>
          </Button>
        </View>
      </Popover>
    );
  }

  saveImg = () => {
    this.setState({
      saveImg: true,
    });
    this.props.onClose();
  };

  save = event => {
    this.props.updateImage(event.nativeEvent.uri);
  };

  resetState = () => {
    this.setState({
      saveImg: false,
    });
  };

  updateValue = type => value => {
    this.setState({
      [type]: value,
    });
  };

  resetAdjustment = () => {
    this.setState({
      brightness: imageAdjusts[0].defaultValue,
      saturation: imageAdjusts[1].defaultValue,
      contrast: imageAdjusts[2].defaultValue,
    });
  };
}

const styles = StyleSheet.create({
  button: {
    padding: 10,
  },
  buttonsContainer: {
    flex: 1,
    flexDirection: 'row',
    width: '90%',
    justifyContent: 'space-evenly',
  },
  buttonText: {
    color: 'white',
  },
  image: {
    height: '100%',
    width: '100%',
    resizeMode: 'contain',
  },
  imageContainer: {
    flex: 3,
    width: '100%',
    paddingHorizontal: 5,
    paddingTop: 5,
    position: 'relative',
  },
  popover: {
    width: Dimensions.get('screen').width * 0.9,
    height: Dimensions.get('screen').height * 0.9,
    justifyContent: 'flex-start',
    alignItems: 'center',
  },
  resetButton: {
    position: 'absolute',
    right: 10,
    bottom: 10,
    backgroundColor: mainThemeColor(1),
    borderRadius: 30,
    justifyContent: 'center',
    alignItems: 'center',
    height: 60,
    width: 60,
    zIndex: 99,
  },
  settingsContainer: {
    marginTop: 10,
    flex: 3,
    justifyContent: 'center',
  },
});
