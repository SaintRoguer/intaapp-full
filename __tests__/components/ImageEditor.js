/**
 * @format
 */
import 'react-native';
import React from 'react';
import {ImageEditor} from '../../src/shared/components/ImageEditor';
import renderer from 'react-test-renderer';

const mockImage = {
  getSource: () => {
    return '';
  },
  getWidth: () => {
    return 0;
  },
  getHeight: () => {
    return 0;
  },
};

describe('ImageEditor component', () => {
  test.skip('renders correctly', () => {
    const component = renderer.create(<ImageEditor image={mockImage} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
