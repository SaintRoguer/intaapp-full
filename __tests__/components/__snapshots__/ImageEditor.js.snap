// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageEditor component renders correctly 1`] = `
<Modal
  hardwareAccelerated={true}
  onDismiss={[Function]}
  onRequestClose={[Function]}
  onShow={[Function]}
  supportedOrientations={
    Array [
      "portrait",
      "portrait-upside-down",
      "landscape",
    ]
  }
  transparent={true}
  visible={false}
>
  <View
    pointerEvents="box-none"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": -2668,
        },
        Object {
          "left": 0,
        },
      ]
    }
  >
    <View
      forwardedRef={[Function]}
      onLayout={[Function]}
      pointerEvents="none"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 20,
          "position": "absolute",
          "right": 0,
          "top": 2668,
        }
      }
    >
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onLayout={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      />
    </View>
    <View
      pointerEvents="box-none"
      style={
        Object {
          "backgroundColor": "transparent",
          "bottom": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": -2668,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.5)",
            "bottom": 2668,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      />
      <View
        pointerEvents="box-none"
        style={
          Object {
            "left": 0,
            "top": 0,
          }
        }
      >
        <View
          forwardedRef={[Function]}
          onLayout={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderBottomColor": "#333438",
              "borderRadius": 3,
              "height": 1200.6000000000001,
              "justifyContent": "flex-start",
              "maxHeight": undefined,
              "maxWidth": undefined,
              "overflow": "hidden",
              "position": "absolute",
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.8,
              "shadowRadius": 2,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
                Object {
                  "scale": 0,
                },
                Object {
                  "perspective": 1000,
                },
              ],
              "width": 675,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 3,
                "paddingHorizontal": 5,
                "paddingTop": 5,
                "position": "relative",
                "width": "100%",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(31, 47, 51, 1)",
                  "borderRadius": 30,
                  "bottom": 10,
                  "height": 60,
                  "justifyContent": "center",
                  "opacity": 1,
                  "position": "absolute",
                  "right": 10,
                  "width": 60,
                  "zIndex": 99,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    Array [
                      Object {
                        "color": "#000",
                        "fontSize": 30,
                      },
                      Object {
                        "color": "#000",
                        "fontSize": 30,
                      },
                      Object {
                        "color": "white",
                        "fontSize": 32,
                      },
                    ],
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                type="Ionicons"
              >
                
              </Text>
            </View>
            <CustomImage
              brightness={0}
              contrast={1}
              onSave={[Function]}
              resizeMode="contain"
              saturation={1}
              saveImage={false}
              src={
                Object {
                  "height": 0,
                  "source": Array [
                    Object {
                      "uri": "",
                    },
                  ],
                  "width": 0,
                }
              }
              style={
                Object {
                  "height": "100%",
                  "resizeMode": "contain",
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "flex": 3,
                "justifyContent": "center",
                "marginTop": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                }
              }
            >
              <Text>
                Brillo
              </Text>
              <RNCSlider
                disabled={false}
                enabled={true}
                inverted={false}
                maximumValue={100}
                minimumTrackTintColor="rgba(31, 47, 51, 1)"
                minimumValue={-100}
                onChange={[Function]}
                onRNCSliderSlidingComplete={null}
                onRNCSliderSlidingStart={null}
                onRNCSliderValueChange={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                step={0}
                style={
                  Array [
                    Object {
                      "height": 40,
                    },
                    Object {
                      "height": 50,
                      "width": 300,
                    },
                  ]
                }
                thumbImage={null}
                thumbTintColor="rgba(31, 47, 51, 1)"
                value={0}
              />
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                }
              }
            >
              <Text>
                Saturación
              </Text>
              <RNCSlider
                disabled={false}
                enabled={true}
                inverted={false}
                maximumValue={2}
                minimumTrackTintColor="rgba(31, 47, 51, 1)"
                minimumValue={0}
                onChange={[Function]}
                onRNCSliderSlidingComplete={null}
                onRNCSliderSlidingStart={null}
                onRNCSliderValueChange={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                step={0}
                style={
                  Array [
                    Object {
                      "height": 40,
                    },
                    Object {
                      "height": 50,
                      "width": 300,
                    },
                  ]
                }
                thumbImage={null}
                thumbTintColor="rgba(31, 47, 51, 1)"
                value={1}
              />
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                }
              }
            >
              <Text>
                Contraste
              </Text>
              <RNCSlider
                disabled={false}
                enabled={true}
                inverted={false}
                maximumValue={1.5}
                minimumTrackTintColor="rgba(31, 47, 51, 1)"
                minimumValue={0.5}
                onChange={[Function]}
                onRNCSliderSlidingComplete={null}
                onRNCSliderSlidingStart={null}
                onRNCSliderValueChange={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                step={0}
                style={
                  Array [
                    Object {
                      "height": 40,
                    },
                    Object {
                      "height": 50,
                      "width": 300,
                    },
                  ]
                }
                thumbImage={null}
                thumbTintColor="rgba(31, 47, 51, 1)"
                value={1}
              />
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "space-evenly",
                "width": "90%",
              }
            }
          >
            <View
              accessible={true}
              focusable={false}
              forwardedRef={[Function]}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#a9a9a9",
                  "borderBottomWidth": null,
                  "borderColor": "#007aff",
                  "borderLeftWidth": null,
                  "borderRadius": 5,
                  "borderRightWidth": null,
                  "borderTopWidth": null,
                  "borderWidth": undefined,
                  "elevation": 2,
                  "flexDirection": "row",
                  "height": 45,
                  "justifyContent": "space-between",
                  "opacity": 1,
                  "padding": 10,
                  "paddingBottom": 6,
                  "paddingTop": 6,
                  "shadowColor": undefined,
                  "shadowOffset": undefined,
                  "shadowOpacity": undefined,
                  "shadowRadius": undefined,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                  }
                }
              >
                Cancelar
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              forwardedRef={[Function]}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#007aff",
                  "borderBottomWidth": null,
                  "borderColor": "#007aff",
                  "borderLeftWidth": null,
                  "borderRadius": 5,
                  "borderRightWidth": null,
                  "borderTopWidth": null,
                  "borderWidth": undefined,
                  "elevation": 2,
                  "flexDirection": "row",
                  "height": 45,
                  "justifyContent": "space-between",
                  "opacity": 1,
                  "padding": 10,
                  "paddingBottom": 6,
                  "paddingTop": 6,
                  "shadowColor": undefined,
                  "shadowOffset": undefined,
                  "shadowOpacity": undefined,
                  "shadowRadius": undefined,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                  }
                }
              >
                Confirmar
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
